### PR TITLE
FeatureFlags implementation

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

File: controllers/cloud.redhat.com/object/object.go [hunk 1]
```diff
@@ -14,4 +14,5 @@ type ClowdObject interface {
 	GetClowdName() string
 	GetUID() types.UID
 	GetDeploymentStatus() *common.DeploymentStatus
+	GetName() string
```
Comment (psav) [inline controllers/cloud.redhat.com/object/object.go:17]: This needs to be removed

File: controllers/cloud.redhat.com/providers/database/localdb_test.go [hunk 2]
```diff
@@ -122,14 +123,16 @@ func TestLocalDBDeployment(t *testing.T) {
...
-	makeLocalDB(&d, nn, &app, &cfg, image, true)
+	provutils.MakeLocalDB(&d, nn, &app, &cfg, image, true, "")
 
 	if d.Spec.Template.Spec.Containers[0].Image != image {
 		t.Fatalf("Image requested %v does not match the one in spec: %v ", image, d.Spec.Template.Spec.Containers[0].Image)
 	}
 	if d.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort != 5432 {
 		t.Fatalf("Port requested %v does not match the one in spec: %v ", 5432, d.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort)
 	}
+	fmt.Printf("\n%v\n", envVars)
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/database/localdb_test.go:134]: These can be removed

File: controllers/cloud.redhat.com/providers/featureflags/localfeatureflags.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+		return nil, err
+	}
+
+	provutils.MakeLocalDBService(&s, nn, p.Env)
+
+	if err = update.Apply(p.Ctx, p.Client, &s); err != nil {
+		return nil, err
+	}
+
+	if p.Env.Spec.Providers.Database.PVC {
```
Comment (psav) [inline controllers/cloud.redhat.com/providers/featureflags/localfeatureflags.go:102]: FeatureFlags!!

File: bundle/tests/scorecard/kuttl/test-kafka-strimzi/01-pods.yaml [hunk 4]
```diff
@@ -19,14 +19,15 @@ spec:
...
     logging:
       mode: none
     objectStore:
-      mode: minio
-      #Possibly need to specify a port here
+      mode: none
     inMemoryDb:
-      mode: redis
+      mode: none
+    featureFlags:
```
Comment (kylape) [inline bundle/tests/scorecard/kuttl/test-kafka-strimzi/01-pods.yaml:29]: can we imply

Comment (kylape) [inline bundle/tests/scorecard/kuttl/test-kafka-strimzi/01-pods.yaml:29]: Can we just omit featureFlags and imply `none` mode?

File: controllers/cloud.redhat.com/config/schema.json [hunk 5]
```diff
@@ -274,6 +277,25 @@
                 "tls"
             ]
         },
+        "FeatureFlagsConfig": {
```
Comment (kylape) [inline controllers/cloud.redhat.com/config/schema.json:280]: Is there no credential needed to query the feature flag server?

File: controllers/cloud.redhat.com/providers/featureflags/localfeatureflags.go [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+		}
+	}
+
+	return ffp, nil
+}
+
+// CreateDatabase ensures a database is created for the given app.  The
+// namespaced name passed in must be the actual name of the db resources
+func (ff *localFeatureFlagsProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error {
+	// ff.Config.Hostname = fmt.Sprintf("%v-featureflags.%v.svc", ff.Env.Name, ff.Env.Status.TargetNamespace)
```
Comment (kylape) [inline controllers/cloud.redhat.com/providers/featureflags/localfeatureflags.go:123]: Should we remove these two commented lines?

File: docs/appconfig/schema-definitions-featureflagsconfig-properties.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Untitled undefined type in AppConfig Schema
```
Comment (kylape) [inline docs/appconfig/schema-definitions-featureflagsconfig-properties.md:1]: this looks weird

**Review by kylape — CHANGES_REQUESTED** [review]
* I don't see an app-interface feature flag impl.  Can we at least stub one out?
* I don't see the featureFlag boolean from the app spec used anywhere.  Did I miss it?

File: docs/appconfig/schema-definitions-featureflagsconfig-properties.md (hunk 7 above)
Comment (psav) [reply docs/appconfig/schema-definitions-featureflagsconfig-properties.md:1]: I don't think we actually use these files - but I will take a short look

File: controllers/cloud.redhat.com/config/schema.json (hunk 5 above)
Comment (psav) [reply controllers/cloud.redhat.com/config/schema.json:280]: Not currently

**Pull request merged**